release/models/network-instance/openconfig-network-instance-l2.yang

```diff
@@ -397,4 +405,314 @@ submodule openconfig-network-instance-l2 {
       }
     }
   }
+
+  grouping l2ni-l2rib-top {
+    description
+      "Top-level grouping for l2rib MAC and MAC-IP table list";
+
+    container l2rib {
+      config false;
+      description
+        "Operational state container for MAC address and MAC-IP address
+         information that is learned and installed into the MAC VRF Layer 2
+         Routing Information Base (L2RIB)";
+
+      container mac-table {
+        description
+          "Operational state container for MAC address information installed
+           into the MAC VRF of the L2RIB";
+
+        container entries {
+          description
+            "Enclosing container for list of MAC address entries";
+
+          list entry {
+            key "mac-address vlan";
+            description "List of learned MAC addresses";
+
+            leaf mac-address {
+              type leafref {
+                path "../state/mac-address";
+              }
+              description "Leafref of MAC address object";
+            }
+
+            leaf vlan {
```

Comment:
@chula-benny We want to remove `vlan` here as a key and just make it a property.  The vlan (mac-vrf) for these entries is specified by the `network-instance` name so it does not make sense to include vlan as a key here.

To query this information as the model is currently written the xpath would like this:

```yang
/network-instances/network-instance[name="vlan55"]/fdb/l2rib/mac-table/entries/entry[vlan="55"][mac-address="5::5"]/state/mac-address 
```

Does not make sense for vlan55 to be a key in two different places in the tree

Reply:
ack, got your email